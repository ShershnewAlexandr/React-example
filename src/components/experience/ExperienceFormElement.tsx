import React from 'react';
import { Grid } from '@material-ui/core';
import { Field } from 'redux-form/lib/immutable';
import { WithTranslation } from 'react-i18next';
import TextInput from '../inputs/TextInput';
import DateInput from '../inputs/DateInput';
import EndDateInputContainer from '../../containers/EndDateInputContainer';

interface ExperienceFormElementProps {
  element: string;
  disabled: boolean;
}

function ExperienceFormElement({
  element,
  disabled,
  t,
}: ExperienceFormElementProps & WithTranslation) {
  return (
    <Grid
      container
      direction="row"
      spacing={4}
      data-testid="experience-form-block"
    >
      <Grid item xs={12}>
        <Field
          name={`${element}.companyName`}
          component={TextInput}
          label={t('experience.companyName')}
          disabled={disabled}
        />
      </Grid>
      <Grid item xs={12}>
        <Field
          name={`${element}.profile`}
          component={TextInput}
          label={t('experience.profile')}
          disabled={disabled}
        />
      </Grid>
      <Grid item xs={6}>
        <Field
          name={`${element}.startYear`}
          component={DateInput}
          label={t('experience.startYear')}
          disabled={disabled}
        />
      </Grid>
      <Grid item xs={6}>
        <Field
          name={`${element}.endYear`}
          component={EndDateInputContainer}
          label={t('experience.endYear')}
          disabled={disabled}
        />
      </Grid>
    </Grid>
  );
}

export default ExperienceFormElement;
