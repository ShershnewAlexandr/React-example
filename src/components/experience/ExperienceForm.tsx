import React from 'react';
import { FieldArray } from 'redux-form/lib/immutable';
import ExperienceFormElement from './ExperienceFormElement';
import DropAddListForm from '../dropAddList/DropAddListForm';
import { ExperienceFormContainerProps } from '../../containers/experience/ExperienceFormContainer';

function ExperienceForm({
  handleSubmit,
  isLoading,
}: ExperienceFormContainerProps) {
  return (
    <form onSubmit={handleSubmit} data-testid="experienceForm">
      <FieldArray
        name="experience"
        component={DropAddListForm}
        formRender={ExperienceFormElement}
        maxElements={10}
        disabled={isLoading}
      />
    </form>
  );
}

export default ExperienceForm;
