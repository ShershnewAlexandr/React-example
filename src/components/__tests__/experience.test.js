import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import ExperienceForm from '../experience/ExperienceForm';
import ExperienceFormElement from '../experience/ExperienceFormElement';

describe('experienceForm', () => {
  const propsForm = {
    handleSubmit: jest.fn(),
    isLoading: false,
  };
  const propsFormElement = {
    element: null,
    disabled: false,
    t: jest.fn(),
  };

  const experienceForm = shallow(<ExperienceForm {...propsForm} />);
  const experienceFormElement = shallow(
    <ExperienceFormElement {...propsFormElement} />,
  );

  it('snapshot Form', () => {
    expect(toJson(experienceForm)).toMatchSnapshot();
  });

  it('should catch submit', () => {
    experienceForm.find('form').simulate('submit');
    expect(propsForm.handleSubmit.mock.calls.length).toBe(1);
  });

  it('snapshot FormElement', () => {
    expect(toJson(experienceFormElement)).toMatchSnapshot();
  });
});
