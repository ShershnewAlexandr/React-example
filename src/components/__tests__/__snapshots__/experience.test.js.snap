// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experienceForm snapshot Form 1`] = `
<form
  data-testid="experienceForm"
  onSubmit={[MockFunction]}
>
  <FieldArray
    component={[Function]}
    disabled={false}
    formRender={[Function]}
    maxElements={10}
    name="experience"
  />
</form>
`;

exports[`experienceForm snapshot FormElement 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  data-testid="experience-form-block"
  direction="row"
  spacing={4}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Field
      component={[Function]}
      disabled={false}
      name="null.companyName"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <Field
      component={[Function]}
      disabled={false}
      name="null.profile"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={6}
  >
    <Field
      component={[Function]}
      disabled={false}
      name="null.startYear"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={6}
  >
    <Field
      component={[Function]}
      disabled={false}
      name="null.endYear"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
